tests/serac_wrapper_tests.cpp

```diff
@@ -210,6 +210,133 @@ TEST_F(WrapperTests, nonlinear_linear_thermal)
   }
 }
 
+TEST_F(WrapperTests, Substitution)
+{
+  // Setup problem
+
+  // Create a coefficient that indicates the x == 0 border of the cube
+  StdFunctionCoefficient x_zero([](const Vector& x) {
+    if (x[0] < 1.e-12) {
+      return 1.;
+    }
+    return 0.;
+  });
+
+  // Create a coefficient that indicates the x == 1 border of the cube
+  StdFunctionCoefficient x_one([](const Vector& x) {
+    if ((1. - x[0]) < 1.e-12) {
+      return 1.;
+    }
+    return 0.;
+  });
+
+  Array<int> bdr_attr_list_zero = serac::makeBdrAttributeList(*pmesh_, x_zero);
+  Array<int> bdr_attr_list_one  = serac::makeBdrAttributeList(*pmesh_, x_one);
+
+  // Set x_zero to be attribute 2 and x_one to be attribute 3
+  Array<int> bdr_attr_list(pfes_->GetNBE());
+  for (int be = 0; be < pfes_->GetNBE(); be++) {
+    bdr_attr_list[be] = (bdr_attr_list_zero[be] - 1) + (bdr_attr_list_one[be] - 1) * 2 + 1;
+  }
+
+  for (int be = 0; be < pfes_->GetNBE(); be++) {
+    pmesh_->GetBdrElement(be)->SetAttribute(bdr_attr_list[be]);
+  }
+
+  // Update attribute data structures
+  pmesh_->SetAttributes();
+
+  Array<int> bdr_attr_is_ess(3);
+  bdr_attr_is_ess[0] = 0;
+  bdr_attr_is_ess[1] = 1;  //< This is an attribute we are looking for
+  bdr_attr_is_ess[2] = 1;  //< This is an attribute we are looking for
+
+  // Get all the essential degrees of freedom
+  Array<int> ess_tdof_list;
+  pfes_->GetEssentialTrueDofs(bdr_attr_is_ess, ess_tdof_list);
+
+  // Boundary conditions evaluation
+  ParGridFunction t_ess(pfes_.get());
+  t_ess = 0.;
+  t_ess.ProjectBdrCoefficient(x_one, bdr_attr_is_ess);
+
+  // Solve Nonlinear
+
+  ConstantCoefficient one(1.);
+
+  auto   diffusion           = std::make_shared<DiffusionIntegrator>(one);
+  auto   nonlinear_diffusion = std::make_unique<BilinearToNonlinearFormIntegrator>(diffusion);
+  double multiplier          = 2.;
+  double offset              = 1.;
+  auto   substitute          = [=](const mfem::Vector& x) {
```

Comment:
Yeah, I think the phrase `substitute` confused me when I first looked at this. I think renaming these fields `transformed` (or something similar) would help a bunch, especially because you're doing a similar thing with the `TransformedScalarCoefficient`s.